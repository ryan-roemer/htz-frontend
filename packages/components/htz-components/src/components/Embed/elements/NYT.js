import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';
import Caption from '../Caption';

NYT.propTypes = {
  inputTemplate: PropTypes.string.isRequired,
  embedType: PropTypes.string.isRequired,
  caption: PropTypes.string,
  credit: PropTypes.string,
  content: PropTypes.string.isRequired,
};

NYT.defaultProps = {
  caption: '',
  credit: '',
};

const videoWrapper = ({ aspectRatio, nyt, }) => {
  const [ width, height, ] = aspectRatio ? aspectRatio.split('/') : [ 16, 9, ];
  const aspect = `${(height / width) * 100}%`;
  const paddingTop = nyt ? '69px' : '';

  return {
    margin: '0',
    paddingBottom: aspect,
    height: '0',
    overflow: 'hidden',
    position: 'relative',
    paddingTop,
  };
};

const VideoWrapper = createComponent(videoWrapper, 'figure');

const videoElement = () => ({
  margin: '0',
  padding: '0',
  height: '100% !important',
  width: '100% !important',
  left: '0',
  top: '0',
  position: 'absolute',
  display: 'block',
  border: 'none',
});

const VideoElement = createComponent(videoElement, 'iframe', props =>
  Object.keys(props)
);

function NYT(props) {
  return (
    <div>
      <VideoWrapper aspectRatio={'16/9'} nyt>
        <VideoElement
          title="New York Times Video - Embed Player"
          width="auto"
          frameBorder="0"
          scrolling="no"
          allowFullScreen="true"
          marginHeight="0"
          marginWidth="0"
          id="nyt_video_player"
          src={props.content}
        />
      </VideoWrapper>
      <Caption
        caption={props.caption}
        credit={props.credit}
        inputTemplate={props.inputTemplate}
        embedType={props.embedType}
      />
    </div>
  );
}

export default NYT;
