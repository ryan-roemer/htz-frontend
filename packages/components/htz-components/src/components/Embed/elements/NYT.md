<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Code Example](#NYT-component)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

### NYT component

An example for a New-York Times video.

```jsx static
<NYT
  content = "https://graphics8.nytimes.com/video/players/offsite/index.html?videoId=100000005256748"
  caption = "video"
  credit = "New-York Times"
/>
```
