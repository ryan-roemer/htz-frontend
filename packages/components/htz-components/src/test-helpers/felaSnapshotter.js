import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
