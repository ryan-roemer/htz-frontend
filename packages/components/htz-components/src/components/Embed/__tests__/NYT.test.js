import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';

import NYT from '../elements/NYT';

it('New-York times ', () => {
  const snapshot = felaSnapshotter(
    <NYT
      source="https://graphics8.nytimes.com/video/players/offsite/index.html?videoId=100000005256748"
      caption="New York Times"
      credit="New York Times"
      embedType="NYTvideo"
      kind="embed"
      inputTemplate="com.polobase.NYTEmbed"
      contentId="7.4028"
      contentName="New York Times"
    />
  );
  expect(snapshot).toMatchSnapshot();
});
