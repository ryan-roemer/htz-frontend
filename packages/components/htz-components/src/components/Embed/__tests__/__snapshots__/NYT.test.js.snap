// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New-York times  1`] = `
Object {
  "component": <div>
    <figure
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <iframe
        allowFullScreen="true"
        className="htz-a htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n"
        frameBorder="0"
        id="nyt_video_player"
        marginHeight="0"
        marginWidth="0"
        onLoad={null}
        scrolling="no"
        src="https://graphics8.nytimes.com/video/players/offsite/index.html?videoId=100000005256748"
        title="New York Times Video - Embed Player"
        width="auto"
      />
    </figure>
  </div>,
  "styles": ".htz-a{margin:0}.htz-b{padding-bottom:56.25%}.htz-c{height:0}.htz-d{overflow:hidden}.htz-e{position:relative}.htz-f{padding-top:69px}.htz-g{padding:0}.htz-h{height:100% !important}.htz-i{width:100% !important}.htz-j{left:0}.htz-k{top:0}.htz-l{position:absolute}.htz-m{display:block}.htz-n{border:none}",
}
`;
